projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -756,6 +707,70 @@ export class IgxDateRangePickerComponent extends DisplayDensityBase
         this.setRequiredToInputs();
     };
 
+    private handleClosing(event: IBaseCancelableBrowserEventArgs): void {
+        if (this.value && !this.value.start && !this.value.end) {
+            this.value = null;
+        }
+
+        const args = { owner: this, cancel: event.cancel, event: event.event };
+        this.closing.emit(args);
+        event.cancel = args.cancel;
+        if (args.cancel) {
+            return;
+        }
+
+        if (this.isDropdown && event.event && !this.element.nativeElement.contains(event.event.target)) {
+            // outside click
+            this.updateValidityOnBlur();
+        } else {
+            // input click
+            if (this.hasProjectedInputs && this._focusedInput) {
+                this._focusedInput.setFocus();
+                this._focusedInput = null;
+            }
+            if (this.inputDirective) {
+                this.inputDirective.focus();
+            }
+        }
+    }
+
+    private subscribeToOverlayEvents() {
+        this._overlayService.onOpening.pipe(
+            filter((overlay) => overlay.id === this._componentID),
```

Comment:
you have same pipe at four places. Please extract this as done in toggleDirective. Something like this:
```
    private _overlaySubFilter: [MonoTypeOperatorFunction<OverlayEventArgs>, MonoTypeOperatorFunction<OverlayEventArgs>] = [
        filter(x => x.id === this._componentID),
        takeUntil(this.destroy$)
    ];
```
and then
```
...
onWhatever.pipe(...this._overlaySubFilter).subscribe
...
```